src/main/kotlin/uk/gov/justice/digital/hmpps/hmppsnonassociationsapi/dto/SyncRequest.kt

```diff
@@ -50,16 +50,17 @@ data class CreateSyncRequest(
     return NonAssociation(
       id = null,
       firstPrisonerNumber = firstPrisonerNumber,
-      firstPrisonerReason = firstPrisonerReason,
+      firstPrisonerRole = firstPrisonerReason.toRole(),
       secondPrisonerNumber = secondPrisonerNumber,
-      secondPrisonerReason = secondPrisonerReason,
-      restrictionType = restrictionType,
-      comment = comment ?: "",
+      secondPrisonerRole = secondPrisonerReason.toRole(),
+      reason = Reason.OTHER, // TODO: what's the "default" reason?
+      restrictionType = restrictionType.toRestrictionType(),
+      comment = comment ?: "No comment provided", // TODO: can we have a better message?
```

Comment:
NB: previously the comment synced to `""` but this is not allowed by DB constraint